### ISSUE TITLE
Aggregation pipeline with date comparison using extended json does not work

### ISSUE THREAD

**Issue by jmisur** [mingo-app/mingo#624]
**Describe the bug**
I'm copy-pasting an aggregation pipeline from spring mongodb log. The pipeline is working correctly in my application (records are found). 
It uses [MongoDB Extended json relaxed date format](https://www.mongodb.com/docs/manual/reference/mongodb-extended-json/#mongodb-bsontype-Date). 

```json
{
  "startedAt": {
    "$gte": {
      "$date": "2023-12-21T11:25:58.344Z"
    }
  }
}

```

However in Mingo, this produces no results for the match stage. 
When I rewrite it to `new Date` format, the results are there. 

```json
{
  "startedAt": {
    "$gte": new Date("2023-12-21T11:25:58.344Z")
  }
}
```

The `startedAt` field in the document is of type `Date`.

**Screenshots**
![Screenshot 2023-12-21 at 13 32 58](https://github.com/mingo-app/mingo/assets/321761/60eb9c76-3f37-4c7f-a163-1c6668bd795d)
![Screenshot 2023-12-21 at 13 33 22](https://github.com/mingo-app/mingo/assets/321761/6515c0e9-61a7-4851-a151-8ccca889a0b0)

**Desktop (please complete the following information):**
 - Mingo version: 1.13.5
 - OS: macOS Sonoma 14.2.1

**Comment by tothradoslav** [comment]
We improved this behaviour as requested. It will convert all extended JSON params into JS before using.

**Comment by tothradoslav** [comment]
This issue is fixed / available in the new release v1.14.1

https://github.com/mingo-app/mingo/releases/tag/v1.14.1

Please reopen this issue if necessary.
